### PR TITLE
Add loading state for profile pages

### LINKED ISSUE #106: Action fields should have placeholders
The placeholders will prevent layout shifts on fetching the profile. I cannot reproduce in this video, but thr first time I logged in on my phone, it took a few seconds to show the data. 

https://user-images.githubusercontent.com/52666982/141650715-5a963ff2-0a2f-4480-b3b9-9d144a1f0f79.mp4

### PR DESCRIPTION
**Pull request by vitokhangnguyen**
Closes #106

### CODE REVIEW

File: src/ui/Loading/animations/FlashingThreeDots..tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  index: number;
+};
+const FlashingDot = styled.span<FlashingDotProps>`
+  ${tw`rounded-full animate-pulse`}
+  animation-duration: 1250ms;
+  animation-delay: ${(p) => p.index * 150}ms;
+`;
+
+const getGapBySize = (size?: LoadingAnimationProps['size']) => {
+  if (size === 'sm') {
```
Comment (Andrewnt219) [inline src/ui/Loading/animations/FlashingThreeDots..tsx:34]: These should be switch

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]

File: src/ui/Loading/animations/FlashingThreeDots..tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { VFC } from 'react';
+import tw, { styled } from 'twin.macro';
+import { LoadingAnimationProps } from '..';
+
+const FlashingThreeDots: VFC<LoadingAnimationProps> = (props) => (
```
Comment (Andrewnt219) [inline src/ui/Loading/animations/FlashingThreeDots..tsx:5]: Can we name this LoadingIndicator?

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]

File: src/ui/Loading/index.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { VFC } from 'react';
+import FlashingThreeDots from './animations/FlashingThreeDots.';
+
+export type LoadingAnimationProps = {
+  variant?: 'primary' | 'secondary';
```
Comment (Andrewnt219) [inline src/ui/Loading/index.tsx:5]: This is more like "color" or "theme". And "animation" is "variant".

**Review by Andrewnt219 — CHANGES_REQUESTED** [review]

File: src/ui/Loading/animations/FlashingThreeDots..tsx (hunk 1 above)
Comment (vitokhangnguyen) [reply src/ui/Loading/animations/FlashingThreeDots..tsx:34]: `switch`looks pretty bad with `return` in them, `if` is more elegant in this case

```js
  if (size === 'sm') {
    return tw`gap-1`;
  }
  if (size === 'lg') {
    return tw`gap-3`;
  }
  return tw`gap-2`;
```

vs 

```js
switch (size) {
    case 'sm':
        return tw`gap-1`;
    case 'lg':
        return tw`gap-3`;
    default:
        return tw`gap-2`;
}
```

Comment (Andrewnt219) [reply src/ui/Loading/animations/FlashingThreeDots..tsx]: individual `if` doesn't hint that they are related to each other. 
```ts
if (a === 'a') return;
if (b === 'b') return;
return;
```

`switch` indicates really clear that we are switching on the same variable. Why is it bad though, syntax-wise they are pretty much the same?

Comment (Andrewnt219) [reply src/ui/Loading/animations/FlashingThreeDots..tsx]: If the condition is based on two or more variables, then `if` guard is better.

Comment (vitokhangnguyen) [reply src/ui/Loading/animations/FlashingThreeDots..tsx]: I think they're ok, it all comes down to coding style I guess. Less keyword means less brain cell needed for me.

And yea they're the same, any half-decent programmer would be about to tell what they do exactly. If they need some kind of hint, they probably need some more skills lol

**Review by Andrewnt219 — APPROVED** [review]

**Pull request merged**